crates/astria-sequencer/src/ibc/ics20_transfer.rs

```diff
@@ -363,14 +416,10 @@ async fn execute_ics20_transfer<S: StateWriteExt>(
                     ))?,
             )
             .context("failed to update escrow account balance in execute_ics20_transfer")?;
+
         state
-            .put_account_balance(
-                recipient,
-                denom.id(),
-                user_balance
-                    .checked_add(packet_amount)
-                    .ok_or(anyhow::anyhow!("overflow when adding to user balance"))?,
-            )
+            .increase_balance(recipient, denom.id(), packet_amount)
```

Comment:
I like these methods, they signal the intent much better.

Can we prevent direct access to `state` and instead provide a wrapper that only exposes methods like `increase_balance`?

Reply:
yeah, it's kind of out of scope for this PR but can do in a follow up

Reply:
Yeah, that makes sense. I would love that